src/EnergyPlus/UnitarySystem.cc

```diff
@@ -11538,28 +11538,29 @@ namespace UnitarySystems {
         } break;
         case HVAC::CoilDX_Cooling: { // CoilCoolingDX
             bool const singleMode = (this->m_SingleMode == 1);
+            CoilPLR = 0.0;
             if (this->m_ControlType == UnitarySysCtrlType::Setpoint) {
                 if (CompressorOn == HVAC::CompressorOp::On) {
-                    CoilPLR = (this->m_CoolingSpeedNum > 1) ? 1.0 : PartLoadRatio;
-                } else
-                    CoilPLR = 0.0;
+                    CoilPLR = (this->m_CoolingSpeedNum > 1) ? m_CoolingSpeedRatio : PartLoadRatio;
+                }
             } else {
                 if (this->m_EMSOverrideCoilSpeedNumOn) {
                     CoilPLR = this->m_CoolingSpeedRatio;
                 } else {
                     if (state.dataUnitarySystems->CoolingLoad) {
-                        if (this->m_CoolingSpeedNum > 1) {
-                            if (!singleMode) {
-                                CoilPLR = (CompressorOn == HVAC::CompressorOp::On) ? 1.0 : 0.0;
-                                this->m_CoolingSpeedRatio = (CompressorOn == HVAC::CompressorOp::On) ? PartLoadRatio : 0.0;
+                        if (CompressorOn == HVAC::CompressorOp::Off &&
+                            this->m_CoolingSpeedNum > 1) { // NOTE: Cooling speed 0 should behave the same as speed 1, but doesn't, and must be
+                                                           // allowed to pass into the simulation code
+                            this->m_CoolingSpeedNum = 1;   // Bypass mixed-speed calculations in called functions
+                        } else {
+                            if (singleMode) {
+                                CoilPLR = (this->m_CoolingSpeedNum == 1)
+                                              ? PartLoadRatio
+                                              : 1.0; // singleMode allows cycling, but not part load operation at higher speeds
```

Comment:
@tanaya-mankad there may be a logic issue here? At line 11551 there are 2 conditionals, CompressorOp::Off && this->m_CoolingSpeedNum > 1. So if CompressorOp::Offand this->m_CoolingSpeedNum == 1 this code will drop to the else where the compressor is On..

Reply:
Thanks @rraustad for finding that fallacy! Unfortunately it doesn't seem to affect the result. I'm attaching a couple of comparison plots - the top plot is the Max Absolute Difference, with red showing the difference on the left scale. The bottom plot is the Max Relative Difference, with red pointing to the right scale. The location is always on Jan 1. 
(The second png is just zoomed way in.) In case it gives us any more info.
![Figure_1](https://github.com/user-attachments/assets/2560c88a-07ca-4e03-9846-0e3acf3bede6)
![Figure_2](https://github.com/user-attachments/assets/ebac51a3-4e20-43d5-818b-ba11d84f89db)

Reply:
Which example file is this?

Reply:
UnitarySystem_MultiSpeedDX_EconoStaging.idf
The regressions are [here ](https://github.com/NREL/EnergyPlus/actions/runs/1224513017)(please don't mind the build problems, apparently it's due to this being a branch on the Big Ladder fork.)

Reply:
That must be a local test file? since it's not in the repo testfile folder. Can you upload it here so I can run it?

Reply:
Of course, and thank you. (It was in the testfiles subdirectory for me.)
[UnitarySystem_MultiSpeedDX_EconoStaging.zip](https://github.com/user-attachments/files/18097936/UnitarySystem_MultiSpeedDX_EconoStaging.zip)

Reply:
I think I figured it out. The BigLadder develop branch is behind NREL develop. When I pulled NREL develop into the BigLadder branch that file showed up.

Reply:
Now I see the same thing you see. The question is if this is due to the changes you made. It's so small it's almost not worth looking into, almost.

![image](https://github.com/user-attachments/assets/5afc172e-8123-4952-a5bb-9a7a34e9505c)

Reply:
Yes, it's tiny! I'd defer to your judgement on whether it's significant. I'm not sure what counts as a "big diff" in regression testing, but 10e-5 relative difference during warmup seems quite difficult to find.

Reply:
This example file came from #9987 and is the only file with local diffs. I think these diffs come from a different OA flow rate at the beginning of the simulation, and for all 4 systems in this file, but I can't see why changing the call to the coil model would affect the calculation for economizer PLR. There are no eio diffs and no warmup diffs but diffs in the eso so it is operational.

![image](https://github.com/user-attachments/assets/b78eef8f-ecc7-4232-b015-d28c47225db5)